### ISSUE TITLE
Build from source Failed

### ISSUE THREAD

**Issue by austindsouza** [eclipse/kura#1394]
Hi I am trying to build Kura from source with a single build step

$ ./build-all.sh

[INFO] Reactor Summary:
[INFO] 
[INFO] target-platform .................................... SUCCESS [  5.113 s]
[INFO] Camel :: Core :: Import sun.misc ................... SUCCESS [  5.195 s]
[INFO] System extension fragment for 'sun.misc' ........... SUCCESS [  0.277 s]
[INFO] com.codeminders.hidapi-parent ...................... SUCCESS [  0.078 s]
[INFO] Java API for working with Human Interface USB Devices (HID) FAILURE [ 11.188 s]
[INFO] Native libraries for 'com.codeminders.hidapi' on ARM 6 HF SKIPPED
[INFO] Native libraries for 'com.codeminders.hidapi' on x86_64 SKIPPED
[INFO] usb4java ........................................... SKIPPED
[INFO] usb4java-javax ..................................... SKIPPED
[INFO] org.eclipse.soda.dk.comm-parent .................... SKIPPED
[INFO] Serial Device based on SODA DK comm ................ SKIPPED
[INFO] Native libraries for 'org.eclipse.soda.dk.comm' on ARM 6 HF SKIPPED
[INFO] Native libraries for 'org.eclipse.soda.dk.comm' on x86_64 SKIPPED
[INFO] jdk.dio-parent ..................................... SKIPPED
[INFO] jdk.dio ............................................ SKIPPED
[INFO] jdk.dio.armv6hf .................................... SKIPPED
[INFO] jdk.dio.x86_64 ..................................... SKIPPED
[INFO] gwt-user ........................................... SKIPPED
[INFO] moquette-broker .................................... SKIPPED
[INFO] p2-repo-common ..................................... SKIPPED
[INFO] p2-repo-equinox_3.11.1 ............................. SKIPPED
[INFO] p2-repo-tests-deps ................................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 28.623 s
[INFO] Finished at: 2017-05-25T12:25:12+05:30
[INFO] Final Memory: 44M/429M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.felix:maven-bundle-plugin:2.3.5:install (default-install) on project com.codeminders.hidapi: Execution default-install of goal org.apache.felix:maven-bundle-plugin:2.3.5:install failed: A required class was missing while executing org.apache.felix:maven-bundle-plugin:2.3.5:install: org/xmlpull/v1/XmlPullParser
[ERROR] -----------------------------------------------------
[ERROR] realm =    extension>org.apache.felix:maven-bundle-plugin:2.3.5
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/root/.m2/repository/org/apache/felix/maven-bundle-plugin/2.3.5/maven-bundle-plugin-2.3.5.jar
[ERROR] urls[1] = file:/root/.m2/repository/biz/aQute/bndlib/1.43.0/bndlib-1.43.0.jar
[ERROR] urls[2] = file:/root/.m2/repository/org/apache/felix/org.apache.felix.bundlerepository/1.6.6/org.apache.felix.bundlerepository-1.6.6.jar
[ERROR] urls[3] = file:/root/.m2/repository/org/osgi/org.osgi.core/4.1.0/org.osgi.core-4.1.0.jar
[ERROR] urls[4] = file:/root/.m2/repository/org/easymock/easymock/2.4/easymock-2.4.jar
[ERROR] urls[5] = file:/root/.m2/repository/org/apache/maven/reporting/maven-reporting-api/2.0.7/maven-reporting-api-2.0.7.jar
[ERROR] urls[6] = file:/root/.m2/repository/org/apache/maven/doxia/doxia-sink-api/1.0-alpha-7/doxia-sink-api-1.0-alpha-7.jar
[ERROR] urls[7] = file:/root/.m2/repository/junit/junit/3.8.1/junit-3.8.1.jar
[ERROR] urls[8] = file:/root/.m2/repository/commons-cli/commons-cli/1.0/commons-cli-1.0.jar
[ERROR] urls[9] = file:/root/.m2/repository/org/codehaus/plexus/plexus-interactivity-api/1.0-alpha-4/plexus-interactivity-api-1.0-alpha-4.jar
[ERROR] urls[10] = file:/root/.m2/repository/org/apache/maven/maven-archiver/2.4.1/maven-archiver-2.4.1.jar
[ERROR] urls[11] = file:/root/.m2/repository/org/codehaus/plexus/plexus-archiver/1.0/plexus-archiver-1.0.jar
[ERROR] urls[12] = file:/root/.m2/repository/org/codehaus/plexus/plexus-io/1.0/plexus-io-1.0.jar
[ERROR] urls[13] = file:/root/.m2/repository/org/codehaus/plexus/plexus-interpolation/1.13/plexus-interpolation-1.13.jar
[ERROR] urls[14] = file:/root/.m2/repository/org/apache/maven/shared/maven-dependency-tree/1.2/maven-dependency-tree-1.2.jar
[ERROR] urls[15] = file:/root/.m2/repository/org/codehaus/plexus/plexus-utils/2.1/plexus-utils-2.1.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
[ERROR] 
[ERROR] -----------------------------------------------------: org.xmlpull.v1.XmlPullParser
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginContainerException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :com.codeminders.hidapi

**Comment by MMaiero** [comment]
Hi, 
which branch are you building?
Which is the host system where the build is running?
And what are the maven and Java versions?

Thanks.
  Matteo

**Comment by austindsouza** [comment]
Hi Matteo,
My setup environment is like this

Branch : release-3.0.0
Host System : CentOS 7
Java version : 1.8.0_45
Maven : 3.3.9

**Comment by MMaiero** [comment]
Hi, 
I've just built Kura using build-all.sh both on develop and release-3.0.0 branch on a vanilla CentOS 7 system with Maven 3.3.9 without any issue. 
I've also tried a build with maven 3.5.0 and again without issues.
Unfortunately, I could not build with the same Java version you have.

Following, the result of `mvn -version`
`Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T16:41:47+00:00)
Maven home: /opt/maven
Java version: 1.8.0_131, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.131-3.b12.el7_3.x86_64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "3.10.0-327.36.1.el7.x86_64", arch: "amd64", family: "unix"`

Maybe there is a problem in your installation?

Best regards,
Matteo
